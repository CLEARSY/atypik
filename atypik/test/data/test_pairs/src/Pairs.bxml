<?xml version="1.0" encoding="UTF-8" ?>
<Machine xmlns='https://www.atelierb.eu/Formats/bxml' version='1.0' name='Pairs' type='abstraction' position='true' semantic='true' b0check='false'>
<Abstract_Variables>
<Attr>
<Pos l='8' c='1' s='18'/>
</Attr>
<Id value='var' typref='4'>
<Attr>
<Pos l='9' c='5' s='3'/>
</Attr>
</Id>
<Id value='var2' typref='5'>
<Attr>
<Pos l='9' c='10' s='4'/>
</Attr>
</Id>
</Abstract_Variables>
<Invariant>
<Attr>
<Pos l='11' c='1' s='9'/>
</Attr>
<Nary_Pred op='&amp;'>
<Exp_Comparison op=':'>
<Attr>
<Pos l='12' c='9' s='1'/>
</Attr>
<Id value='var' typref='4'>
<Attr>
<Pos l='12' c='5' s='3'/>
</Attr>
</Id>
<Binary_Exp op='*s' typref='6'>
<Attr>
<Pos l='12' c='16' s='1'/>
</Attr>
<Id value='REAL' typref='7'>
<Attr>
<Pos l='12' c='11' s='4'/>
</Attr>
</Id>
<Id value='INTEGER' typref='2'>
<Attr>
<Pos l='12' c='18' s='7'/>
</Attr>
</Id>
</Binary_Exp>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='13' c='10' s='1'/>
</Attr>
<Id value='var2' typref='5'>
<Attr>
<Pos l='13' c='5' s='4'/>
</Attr>
</Id>
<Unary_Exp op='POW' typref='8'>
<Attr>
<Pos l='13' c='12' s='3'/>
</Attr>
<Binary_Exp op='*s' typref='5'>
<Attr>
<Pos l='13' c='24' s='1'/>
</Attr>
<Id value='INTEGER' typref='2'>
<Attr>
<Pos l='13' c='16' s='7'/>
</Attr>
</Id>
<Id value='BOOL' typref='9'>
<Attr>
<Pos l='13' c='26' s='4'/>
</Attr>
</Id>
</Binary_Exp>
</Unary_Exp>
</Exp_Comparison>
</Nary_Pred>
</Invariant>
<Initialisation>
<Attr>
<Pos l='15' c='1' s='14'/>
</Attr>
<Nary_Sub op='||'>
<Assignement_Sub>
<Attr>
<Pos l='16' c='9' s='2'/>
</Attr>
<Variables>
<Id value='var' typref='4'>
<Attr>
<Pos l='16' c='5' s='3'/>
</Attr>
</Id>
</Variables>
<Values>
<Binary_Exp op='|-&gt;' typref='4'>
<Attr>
<Pos l='16' c='16' s='3'/>
</Attr>
<Real_Literal value='1.0' typref='10'>
<Attr>
<Pos l='16' c='12' s='3'/>
</Attr>
</Real_Literal>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='16' c='20' s='1'/>
</Attr>
</Integer_Literal>
</Binary_Exp>
</Values>
</Assignement_Sub>
<Assignement_Sub>
<Attr>
<Pos l='17' c='10' s='2'/>
</Attr>
<Variables>
<Id value='var2' typref='5'>
<Attr>
<Pos l='17' c='5' s='4'/>
</Attr>
</Id>
</Variables>
<Values>
<Nary_Exp op='{' typref='5'>
<Attr>
<Pos l='17' c='13' s='1'/>
</Attr>
<Binary_Exp op='|-&gt;' typref='11'>
<Attr>
<Pos l='17' c='17' s='3'/>
</Attr>
<Integer_Literal value='0' typref='1'>
<Attr>
<Pos l='17' c='15' s='1'/>
</Attr>
</Integer_Literal>
<Boolean_Literal value='FALSE' typref='0'>
<Attr>
<Pos l='17' c='21' s='5'/>
</Attr>
</Boolean_Literal>
</Binary_Exp>
<Binary_Exp op=',' typref='11'>
<Attr>
<Pos l='17' c='31' s='1'/>
</Attr>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='17' c='30' s='1'/>
</Attr>
</Integer_Literal>
<Boolean_Literal value='TRUE' typref='0'>
<Attr>
<Pos l='17' c='33' s='4'/>
</Attr>
</Boolean_Literal>
</Binary_Exp>
</Nary_Exp>
</Values>
</Assignement_Sub>
</Nary_Sub>
</Initialisation>
<TypeInfos>
<Type id='0'>
<Id value='BOOL'/>
</Type>
<Type id='1'>
<Id value='INTEGER'/>
</Type>
<Type id='2'>
<Unary_Exp op='POW'>
<Id value='INTEGER'/>
</Unary_Exp>
</Type>
<Type id='3'>
<Unary_Exp op='POW'>
<Unary_Exp op='POW'>
<Id value='INTEGER'/>
</Unary_Exp>
</Unary_Exp>
</Type>
<Type id='4'>
<Binary_Exp op='*'>
<Id value='REAL'/>
<Id value='INTEGER'/>
</Binary_Exp>
</Type>
<Type id='5'>
<Unary_Exp op='POW'>
<Binary_Exp op='*'>
<Id value='INTEGER'/>
<Id value='BOOL'/>
</Binary_Exp>
</Unary_Exp>
</Type>
<Type id='6'>
<Unary_Exp op='POW'>
<Binary_Exp op='*'>
<Id value='REAL'/>
<Id value='INTEGER'/>
</Binary_Exp>
</Unary_Exp>
</Type>
<Type id='7'>
<Unary_Exp op='POW'>
<Id value='REAL'/>
</Unary_Exp>
</Type>
<Type id='8'>
<Unary_Exp op='POW'>
<Unary_Exp op='POW'>
<Binary_Exp op='*'>
<Id value='INTEGER'/>
<Id value='BOOL'/>
</Binary_Exp>
</Unary_Exp>
</Unary_Exp>
</Type>
<Type id='9'>
<Unary_Exp op='POW'>
<Id value='BOOL'/>
</Unary_Exp>
</Type>
<Type id='10'>
<Id value='REAL'/>
</Type>
<Type id='11'>
<Binary_Exp op='*'>
<Id value='INTEGER'/>
<Id value='BOOL'/>
</Binary_Exp>
</Type>
</TypeInfos>

</Machine>