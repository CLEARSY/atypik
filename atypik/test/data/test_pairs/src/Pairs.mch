/* Pairs
 * Author: vdemeulenaere
 * Creation date: 23/05/2023
 */
MACHINE
    Pairs

ABSTRACT_VARIABLES 
    var, var2
    
INVARIANT
    var : REAL * INTEGER &
    var2 : POW(INTEGER * BOOL)
    
INITIALISATION
    var := 1.0 |-> 1 ||
    var2 := {(0 |-> FALSE), (1, TRUE)}

END
