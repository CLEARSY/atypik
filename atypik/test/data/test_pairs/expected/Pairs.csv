Expression	Identifier	Line	Column	Type
"TRUE"	20	17	33	BOOL
"1"	19	17	30	INTEGER
"(0) |-> (FALSE)"	18	17	17	(PRODUCT INTEGER BOOL)
"var"	0	16	5	(PRODUCT REAL INTEGER)
"{(0) |-> (FALSE),(1) , (TRUE)}"	22	17	13	(POW (PRODUCT INTEGER BOOL))
"var2"	1	17	5	(POW (PRODUCT INTEGER BOOL))
"1"	14	16	20	INTEGER
"REAL"	2	12	11	(POW REAL)
"POW((INTEGER) *s (BOOL))"	11	13	12	(POW (POW (PRODUCT INTEGER BOOL)))
"INTEGER"	3	12	18	(POW INTEGER)
"0"	16	17	15	INTEGER
"(1.0) |-> (1)"	15	16	16	(PRODUCT REAL INTEGER)
"(1) , (TRUE)"	21	17	31	(PRODUCT INTEGER BOOL)
"(INTEGER) *s (BOOL)"	8	13	24	(POW (PRODUCT INTEGER BOOL))
"(REAL) *s (INTEGER)"	4	12	16	(POW (PRODUCT REAL INTEGER))
"1.0"	13	16	12	REAL
"BOOL"	7	13	26	(POW BOOL)
"FALSE"	17	17	21	BOOL
